openshift-ci/deploy_metallb.sh

```diff
@@ -17,16 +17,107 @@ if [ ! -d ./metallb-operator ]; then
 	git checkout ${METALLB_OPERATOR_BRANCH}
 	cd -
 fi
-cd metallb-operator
 
-# install yq v4 for metallb deployment
-go install -mod='' github.com/mikefarah/yq/v4@v4.13.3
+rm -rf metallb-operator-deploy/manifests
+rm -rf metallb-operator-deploy/bundle
+rm metallb-operator-deploy/bundleci.Dockerfile
 
-yq e --inplace '.spec.template.spec.containers[0].env[] |= select (.name=="SPEAKER_IMAGE").value|="'${METALLB_IMAGE_BASE}':'${METALLB_IMAGE_TAG}'"' ${metallb_dir}/metallb-operator-deploy/controller_manager_patch.yaml
-yq e --inplace '.spec.template.spec.containers[0].env[] |= select (.name=="CONTROLLER_IMAGE").value|="'${METALLB_IMAGE_BASE}':'${METALLB_IMAGE_TAG}'"' ${metallb_dir}/metallb-operator-deploy/controller_manager_patch.yaml
-yq e --inplace '.spec.template.spec.containers[0].env[] |= select (.name=="FRR_IMAGE").value|="'${METALLB_IMAGE_BASE}':'${FRR_IMAGE_TAG}'"' ${metallb_dir}/metallb-operator-deploy/controller_manager_patch.yaml
+cp metallb-operator/bundleci.Dockerfile metallb-operator-deploy 
+cp -r metallb-operator/manifests/ metallb-operator-deploy/manifests 
+cp -r metallb-operator/bundle/ metallb-operator-deploy/bundle 
 
-PATH="${GOPATH}:${PATH}" ENABLE_OPERATOR_WEBHOOK=true KUSTOMIZE_DEPLOY_DIR="../metallb-operator-deploy" IMG="${METALLB_IMAGE_BASE}:${METALLB_OPERATOR_IMAGE_TAG}" make deploy
+cd metallb-operator-deploy
+
+find . -type f -name "*clusterserviceversion*.yaml" -exec sed -i 's/quay.io\/openshift\/origin-metallb:.*$/'${METALLB_IMAGE_BASE}':'${METALLB_IMAGE_TAG}'/g' {} +
+find . -type f -name "*clusterserviceversion*.yaml" -exec sed -i 's/quay.io\/openshift\/origin-metallb-frr:.*$/'${METALLB_IMAGE_BASE}':'${FRR_IMAGE_TAG}'/g' {} +
+find . -type f -name "*clusterserviceversion*.yaml" -exec sed -i 's/quay.io\/openshift\/origin-metallb-operator:.*$/'${METALLB_IMAGE_BASE}':'${METALLB_OPERATOR_IMAGE_TAG}'/g' {} +
+find . -type f -name "*clusterserviceversion*.yaml" -exec sed -r -i 's/name: metallb-operator\.v4\.11\.0/name: metallb-operator.v0.0.0/g' {} +
```

Comment:
im wondering if we can somehow keep the 4.11 version in a var for future update (i guess it's too complicated with sed).

Reply:
Ah dang, this is totally wrong, good catch. We should not depend on the version, I should be more aggressive in sed-ing and replace any version with 0.0.0 which is the main point of it, so it doesn't depend on the version.